[ModuleInterface] Look for a prebuilt module by full target name

jrose-apple opened the pull request:
Although the default cache location separates by platform, a *combined* cache is still an interesting idea, and in that case we need to use the full platform names.

rdar://problem/50371801

jrose-apple commented:
@swift-ci Please smoke test

jrose-apple commented:
@swift-ci Please smoke test

harlanhaskins commented on lib/Frontend/ParseableInterfaceModuleLoader.cpp:
```diff
@@ -926,16 +969,29 @@ class ParseableInterfaceModuleLoaderImpl {
+      } else {
+        // Hack: deal with prebuilds of modules that still use the target-based
+        // names. This code is duplicated from above but there doesn't seem to
+        // be a great way to factor it out due to the early return.
```
Would
```c++
if (!path)
  path = computeFallbackPrebuiltModulePath(scratch)
```
work?

jrose-apple replied:
…

Yes, it would. I wrote this against a version that didn't check `fs::exists` when computing the path, and then hastily revised it once I saw that that got in. I'll change it to that.

jrose-apple commented:
@swift-ci Please smoke test

jrose-apple committed:
[ModuleInterface] Look for a prebuilt module by full target name

Although the default cache location separates by platform, a
/combined/ cache is still an interesting idea, and in that case we
need to use the full platform names.

jrose-apple commented:
@swift-ci Please smoke test
